src/core/libmaven/mzSample.cpp

```diff
@@ -427,6 +414,36 @@ void mzSample::parseMzMLChromatogromList(xml_node &chromatogramList)
 	}
 }
 
+int mzSample::getSampleNoChromatogram(string chromatogramId) {
+
+	std::regex rxSampleNumber("sample\ *\=\ *([0-9]+)");
+	std::vector<int> results;
+    for(std::sregex_iterator i = std::sregex_iterator(chromatogramId.begin(), chromatogramId.end(), rxSampleNumber);
+            i != std::sregex_iterator(); 
+            ++i) 
+    { 
+        std::smatch m = *i;
+        results.push_back(std::stoi( m[1].str().c_str() ));
+    } 
+
+    for (auto n: results)
+        return n;
```

Comment:
it will always return after the first loop. so we can simply do `return results[0]` 
and if that's the case why do we need a `std::vector<int> results` ?

did you intend to do something else ?

Reply:
@rish9511 

Consider a string, where we are trying to match some regex. There are 2 corner cases we need to take care of: 
- **What if there are two substrings which match the regex** -- Hence there is a vector of results
- **What if there is no substring which matches the regex** -- The for loop for results won't execute and hence it will skip that part.

I agree there will be numerous ways to do this. Let me know if this still confuses you. I'll make changes in the code.

Reply:
@rish9511 We are taking here the assumptions that we only want the first result.